poutyne/framework/callbacks/wandb_logger.py

```diff
@@ -0,0 +1,256 @@
+            wandb_logger.run.log({"an example": image})
+    """
+
+    def __init__(
+        self,
+        *,
+        name: Optional[str] = None,
+        group: Optional[str] = None,
+        config: Optional[Dict] = None,
+        save_dir: Optional[str] = None,
+        offline: Optional[bool] = False,
+        run_id: Optional[str] = None,
+        anonymous: Optional[bool] = None,
+        version: Optional[str] = None,
+        project: Optional[str] = None,
+        experiment=None,
+        batch_granularity: Optional[bool] = False,
+        checkpoints_path: Optional[str] = None,
+        initial_artifacts_paths: Optional[List[str]] = None,
+        log_gradient_frequency: Optional[int] = None,
+        training_batch_shape: Optional[tuple] = None,
+    ) -> None:
+        # pylint: disable-msg=too-many-locals
+
+        super().__init__(batch_granularity=batch_granularity)
+
+        if wandb is None:
+            raise ImportError("WandB needs to be installed to use this callback.")
+
+        anonymous_lut = {True: "allow", False: None}
+        self._wandb_init = dict(
+            name=name,
+            group=group,
+            config=config,
+            project=project,
+            id=version or run_id,
+            dir=save_dir,
+            resume="allow",
+            anonymous=anonymous_lut.get(anonymous, anonymous),
+        )
+
+        if experiment is None:
+
+            if offline:
+                os.environ["WANDB_MODE"] = "dryrun"
+
+            if wandb.run is None:
+                self.run = wandb.init(**self._wandb_init)
+            else:
+                warnings.warn(
+                    "There is already a wandb run experience running. This callback will reuse this run. If you want "
+                    "to start a new one stop this process and call `wandb.finish()` before starting again."
+                )
+                self.run = wandb.run
+        else:
+            self.run = experiment
+
+        self.run.config.update({"run_id": self.run.id})
+        self.log_gradient_frequency = log_gradient_frequency
+        self.training_batch_shape = training_batch_shape
+        self.checkpoints_path = checkpoints_path
+        if initial_artifacts_paths is not None:
+            self._log_artifacts(initial_artifacts_paths, name="Initial-artifacts", artifact_type="Miscellaneous")
+
+    def _watch_gradient(self) -> None:
+        """
+        Activate wandb gradient watching.
+        """
+        self.run.watch(self.model.network, log="all", log_freq=self.log_gradient_frequency)
+
+    def _save_architecture(self) -> None:
+        """
+        Save architecture.
+        """
+        dummies_batch = torch.randn(self.training_batch_shape)
+        save_path = self.run.dir + "/" + self.run.name + "_model.onnx"
+        torch.onnx.export(self.model.network, dummies_batch, save_path)
+        self.run.save(save_path)
+
+    def on_train_begin(self, logs: Dict):
+        super().on_train_begin(logs)
```

Comment:
```suggestion
        self.on_train_begin(logs)
```
Since child of logger, don't need to do super.

Reply:
Nope. Doing that, you just create infinite recursion.